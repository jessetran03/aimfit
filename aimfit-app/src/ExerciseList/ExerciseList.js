import React, { Component } from 'react'
import './ExerciseList.css'
import config from '../config'
import STORE from '../STORE';

export default class ExerciseList extends Component {
  state = {
    store: STORE,
  }
  /*state = {
    exercises: [],
  }

  componentDidMount() {
    this.getData()
  }

  getData = () => {
    Promise.all([
      fetch(`${config.API_ENDPOINT}/exercises`)
    ])
      .then(([exercisesRes]) => {
        if (!exercisesRes.ok)
          return exercisesRes.json().then(e => Promise.reject(e))

        return Promise.all([
          exercisesRes.json(),
        ])
      })
      .then(([exercises]) => {
        this.setState({ exercises })
      })
      .catch(error => {
        console.error({ error })
      })
  }*/

  render() {
    const exercises = this.state.store.exercises;
    return (
      <>
        <section className='exercises'>
          <h2>All Exercises</h2>
          <ul>
            {exercises.map(exercise => (
              <li key={exercise.id}>
                <h3>{exercise.exercise_name}</h3>
                <br />
                <select>
                  <option>Select...</option>
                  <option>Push workout</option>
                  <option>Pull workout</option>
                </select>
                &nbsp;
                <button>
                  Add to workout
                </button>
              </li>
            ))}
          </ul>
        </section>
      </>
    )
  }
}