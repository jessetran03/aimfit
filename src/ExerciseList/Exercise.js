import React, { Component } from 'react'
import config from '../config'
import TokenService from '../services/token-service'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

export default class Exercise extends Component {
  static propTypes = {
    id: PropTypes.number,
    name: PropTypes.string
  }

  handleAddExercise = e => {
    e.preventDefault()
    const newWorkoutExercise = {
      workout_id: this.props.workoutId,
      exercise_id: this.props.id,
    }
    fetch(`${config.API_ENDPOINT}/workout_exercises`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
      body: JSON.stringify(newWorkoutExercise),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(workoutExercise => {
        this.props.handleGoBack()
      })
      .catch(error => {
        console.error({ error })
      })
  };

  render() {
    const { id, name } = this.props
    const path = 'workouts/exercise_log/' + id
    return (
      <li key={id}>
        <Link to={path}>
          <h3>{name}</h3>
        </Link>
      </li>
    )
  }
}