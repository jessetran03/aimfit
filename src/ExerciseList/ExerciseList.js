import React, { Component } from 'react'
import './ExerciseList.css'
import Exercise from './Exercise'
import config from '../config'
export default class ExerciseList extends Component {
  static defaultProps = {
    match: { params: {} },
    location: { pathname: {} }
  }
  state = {
    exercises: [],
    muscle: 'Chest'
  }

  muscles = ['Chest', 'Back', 'Triceps', 'Biceps', 'Shoulders', 'Traps', 'Quads', 'Hamstrings', 'Calves']

  componentDidMount() {
    this.getData()
  }

  getData = () => {
    Promise.all([
      fetch(`${config.API_ENDPOINT}/exercises`)
    ])
      .then(([exercisesRes]) => {
        if (!exercisesRes.ok)
          return exercisesRes.json().then(e => Promise.reject(e))

        return Promise.all([
          exercisesRes.json(),
        ])
      })
      .then(([exercises]) => {
        this.setState({ exercises })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  filterMuscle = e => {
    const muscle = e.target.value;
    this.setState({
      muscle: muscle
    })
  }

  handleGoBack = () => {
    this.props.history.goBack()
  };

  render() {
    const exercises = this.state.exercises;
    const workoutId = this.props.match.params.workoutId
    const muscle = this.state.muscle;
    const muscles = this.muscles;
    return (
      <>
        <h2 className='add-exercise-header'>Exercises</h2>
        <button
          onClick={this.handleGoBack}
        >
          Back
          </button>
        <section className='muscle-group'>
          <h3 className='muscle-header'>Muscle Group</h3>
          <ul>
            {muscles.map(muscle => (
              (muscle === this.state.muscle
                ? <li key={muscle}><button className='selected' onClick={this.filterMuscle} value={muscle}>{muscle}</button></li>
                : <li key={muscle}><button onClick={this.filterMuscle} value={muscle}>{muscle}</button></li>
              )
            ))}
          </ul>
        </section>
        <section className='workouts'>
          <div>
            Showing exercises for {muscle}:
          </div>
          <ul>
            {exercises
              .filter(exercise => exercise.muscle === muscle)
              .map(exercise => (
                <Exercise
                  key={exercise.id}
                  id={exercise.id}
                  name={exercise.exercise_name}
                  workoutId={workoutId}
                  handleGoBack={this.handleGoBack}
                />
              ))}
          </ul>
        </section>
      </>
    )
  }
}