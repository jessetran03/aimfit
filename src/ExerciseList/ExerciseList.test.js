import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import ExerciseList from './ExerciseList'
import renderer from 'react-test-renderer'

it('renders ExerciseList without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <ExerciseList />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render ExerciseList without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <ExerciseList />
    </BrowserRouter>
  )
    .toJSON();
    expect(wrapper).toMatchSnapshot()
  })