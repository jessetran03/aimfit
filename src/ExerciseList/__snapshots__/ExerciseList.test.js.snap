// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExerciseList without data 1`] = `
Array [
  <h2
    className="add-exercise-header"
  >
    Exercises
  </h2>,
  <button
    onClick={[Function]}
  >
    Back
  </button>,
  <section
    className="muscle-group"
  >
    <h3
      className="muscle-header"
    >
      Muscle Group
    </h3>
    <ul>
      <li>
        <button
          className="selected"
          onClick={[Function]}
          value="Chest"
        >
          Chest
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Back"
        >
          Back
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Triceps"
        >
          Triceps
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Biceps"
        >
          Biceps
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Shoulders"
        >
          Shoulders
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Traps"
        >
          Traps
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Quads"
        >
          Quads
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Hamstrings"
        >
          Hamstrings
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
          value="Calves"
        >
          Calves
        </button>
      </li>
    </ul>
  </section>,
  <section
    className="workouts"
  >
    <div>
      Showing exercises for 
      Chest
      :
    </div>
    <ul />
  </section>,
]
`;
